mp2-v1/src/final_extraction/api.rs

```diff
@@ -219,6 +269,38 @@ impl CircuitInput {
         let length_proof = ProofWithVK::deserialize(&length_proof)?;
         Ok(Self::Lengthed(LengthedCircuitInput { base, length_proof }))
     }
+    /// Instantiate inputs for the dummy circuit dealing with no provable extraction case
+    pub fn new_no_provable_input<PrimaryIndex: PartialEq + Eq + Default + Clone + Debug>(
+        block_number: U256,
+        block_hash: OffChainRootOfTrust,
+        prev_block_hash: HashOutput,
```

Comment:
Do we have to provide one ? If it's not equal to the previous insertion, will it fail ? Couldn't we simply put a dummy value inside the API ?

Reply:
Yes, if we don't provide here the block_hash used for previous primary index, a constraint in IVC proof will fail. I think it would be simpler and cleaner to provide the previous root of trust here rather than having to handle that constraint differently in the IVC proof, no?

Reply:
As agreed in DM, in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/415/commits/2f71b25f22f08043f029b17af38adf20bbe76faf I changed the interface of this method to take as input the previous IVC proof rather than the previous block hash, so that we can fetch the previous block hash from the proof internally